src/rhsm/connection.py

```diff
@@ -554,40 +556,48 @@ def _request(self, request_type, method, info=None):
-            log.debug("Using proxy: %s:%s" % (self.proxy_hostname, self.proxy_port))
-            conn = RhsmProxyHTTPSConnection(self.proxy_hostname, self.proxy_port,
-                                            username=self.proxy_user,
-                                            password=self.proxy_password,
-                                            ssl_context=context, timeout=self.timeout)
-            # this connection class wants the full url
-            handler = "https://%s:%s%s" % (self.host, self.ssl_port, handler)
-        else:
-            conn = HTTPSConnection(self.host, self.ssl_port, ssl_context=context, timeout=self.timeout)
+        # Use a retry for intermittent connection issues
+        for num in range(ssl_retries):
+            if self.proxy_hostname and self.proxy_port:
+                log.debug("Using proxy: %s:%s" % (self.proxy_hostname, self.proxy_port))
+                conn = RhsmProxyHTTPSConnection(self.proxy_hostname, self.proxy_port,
+                                                username=self.proxy_user,
+                                                password=self.proxy_password,
+                                                ssl_context=context)
+                # this connection class wants the full url
+                handler = "https://%s:%s%s" % (self.host, self.ssl_port, handler)
+            else:
+                conn = httpslib.HTTPSConnection(self.host, self.ssl_port, ssl_context=context)
 
-        if info is not None:
-            body = json.dumps(info, default=json.encode)
-        else:
-            body = None
+            if info is not None:
+                body = json.dumps(info, default=json.encode)
+            else:
+                body = None
 
-        log.debug("Making request: %s %s" % (request_type, handler))
+            log.debug("Making request: %s %s" % (request_type, handler))
 
-        if self.user_agent:
-            self.headers['User-Agent'] = self.user_agent
+            if self.user_agent:
+                self.headers['User-Agent'] = self.user_agent
 
-        headers = self.headers
-        if body is None:
-            headers = dict(self.headers.items() +
-                           {"Content-Length": "0"}.items())
+            headers = self.headers
+            if body is None:
+                headers = dict(self.headers.items() +
+                               {"Content-Length": "0"}.items())
+
+            try:
+                conn.request(request_type, handler, body=body, headers=headers)
+            except SSLError, se:
+                if self.cert_file:
+                    id_cert = certificate.create_from_file(self.cert_file)
+                    if not id_cert.is_valid():
+                        raise ExpiredIdentityCertException()
+                    # we should retry here, its not an identity cert issue
+                    if num == ssl_retries - 1:
+                        raise se
```

Comment:
If we just want to reraise the current exception use only `raise` instead of `raise se`.  The [problem](http://www.ianbicking.org/blog/2007/09/re-raising-exceptions.html) with the latter is you lose information in the original exception (like the line numbers involved).